$NetBSD: patch-GNUmakefile,v 1.2 2026/01/05 22:10:35 alnsn Exp $

- Install suffixed binaries so multiple Lua versions can coexist.
- Install config file to LUA_EXAMPLESDIR.

--- GNUmakefile.orig	2025-12-28 23:01:55.000000000 +0000
+++ GNUmakefile
@@ -109,8 +109,8 @@ INSTALL_FILES =
 
 install: all install-config
 	mkdir -p '$(DESTDIR)$(bindir)/'
-	$(INSTALL) '$(builddir)/luarocks' '$(DESTDIR)$(bindir)/luarocks'
-	$(INSTALL) '$(builddir)/luarocks-admin' '$(DESTDIR)$(bindir)/luarocks-admin'
+	$(INSTALL) '$(builddir)/luarocks' '$(DESTDIR)$(bindir)/luarocks-$(LUA_VERSION)'
+	$(INSTALL) '$(builddir)/luarocks-admin' '$(DESTDIR)$(bindir)/luarocks-admin-$(LUA_VERSION)'
 	find src/luarocks/ -type d | while read f; \
 	do \
 	   mkdir -p '$(DESTDIR)$(luadir)'/`echo $$f | sed 's,^src/,,'`; \
@@ -131,8 +131,8 @@ ifeq (,$(findstring $(LUA_VERSION),"5.3"
 endif
 
 install-config:
-	mkdir -p '$(DESTDIR)$(luarocksconfdir)/'
-	$(INSTALL_DATA)  '$(builddir)/config-$(LUA_VERSION).lua' '$(DESTDIR)$(luarocksconfdir)/config-$(LUA_VERSION).lua'
+	mkdir -p '$(DESTDIR)$(PREFIX)/$(LUA_EXAMPLESDIR)/'
+	$(INSTALL_DATA)  '$(builddir)/config-$(LUA_VERSION).lua' '$(DESTDIR)$(PREFIX)/$(LUA_EXAMPLESDIR)/config-$(LUA_VERSION).lua'
 
 uninstall:
 	rm -rf $(DESTDIR)$(bindir)/luarocks \
